modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/CommonAnalysisPlugin.java

```diff
@@ -652,4 +654,23 @@ public List<PreConfiguredTokenizer> getPreConfiguredTokenizers() {
 
         return tokenizers;
     }
+
+    @Override
+    public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
+        if (SynonymsManagementAPIService.isEnabled()) {
+            return Collections.singletonList(SynonymsManagementAPIService.getSystemIndexDescriptor());
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public String getFeatureName() {
```

Comment:
Same comment as before - this would probably be better encapsulated into a separate Plugin. We can always add it later of course.